site/app/controllers/forum/ForumController.php

```diff
@@ -147,20 +176,23 @@ public function publishThread(){
         $thread_content = str_replace("\r", "", $_POST["thread_content"]);
         $anon = (isset($_POST["Anon"]) && $_POST["Anon"] == "Anon") ? 1 : 0;
         $announcment = (isset($_POST["Announcement"]) && $_POST["Announcement"] == "Announcement" && $this->core->getUser()->getGroup() < 3) ? 1 : 0 ;
-        $category_id = (int)$_POST["cat"];
+        $categories_ids  = array();
+        foreach ($_POST["cat"] as $category_id) {
+            $categories_ids[] = (int)$category_id;
+        }
         if(empty($title) || empty($thread_content)){
             $this->core->addErrorMessage("One of the fields was empty or bad. Please re-submit your thread.");
             $this->core->redirect($this->core->buildUrl(array('component' => 'forum', 'page' => 'create_thread')));
-        }else if(!$this->isValidCategory($category_id)){
-            $this->core->addErrorMessage("You must select a valid category. Please re-submit your thread.");
+        }else if(!$this->isValidCategories($categories_ids)){
+            $this->core->addErrorMessage("You must select valid categories. Please re-submit your thread.");
```

Comment:
If the user has JavaScript disabled it defaults to all categories for the thread. Seems it doesn't trigger this error message.

Reply:
Thankyou for pointing out.
Fixed it as discussed on slack.